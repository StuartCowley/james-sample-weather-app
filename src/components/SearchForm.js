import React from "react";
import PropTypes from "prop-types";
import "../styles/SearchForm.css";
import getForecast from "../requests/getForecast";

function SearchForm({ searchText, setSearchText, onSubmit }) {
  const handleInputChange = (event) => setSearchText(event.target.value);

  // attempting to add an event listener to use enter as well as mouse click
  const checkKeyPress = (event) => {
    const { keyCode } = event;
    if (keyCode === 13) {
      console.log(searchText);
      getForecast(searchText);
    }
  };

  return (
    <div className="search-form" onSubmit={onSubmit}>
      <input
        id="searchBox"
        type="text"
        onChange={handleInputChange}
        onKeyDown={checkKeyPress}
        value={searchText}
        placeholder="Enter a city..."
      />
      <button id="button" type="submit" onClick={onSubmit}>
        Search
      </button>
    </div>
  );
}

SearchForm.propTypes = {
  searchText: PropTypes.string.isRequired,
  setSearchText: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
};

export default SearchForm;
