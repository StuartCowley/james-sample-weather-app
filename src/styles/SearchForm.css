.search-form {
    text-align: center;
    
}

#searchBox{
    display: flex;
    width: 80%;
    height: 40px;
    font-size: 200%;
    margin-bottom: 10px;
    border-radius: 5px;
    text-align: center;
    font-weight: 400;
    font-variant: small-caps;
    margin-right: auto;
    margin-left: auto;
}

#searchBox:focus {
    border: 3px solid #555;
}


#button {
  background-color: #c2fbd7;
  border-radius: 100px;
  box-shadow: rgba(44, 187, 99, .2) 0 -25px 18px -14px inset,rgba(44, 187, 99, .15) 0 1px 2px,rgba(44, 187, 99, .15) 0 2px 4px,rgba(44, 187, 99, .15) 0 4px 8px,rgba(44, 187, 99, .15) 0 8px 16px,rgba(44, 187, 99, .15) 0 16px 32px;
  color: green;
  cursor: pointer;
  display: inline-block;
  padding: 7px 20px;
  text-align: center;
  text-decoration: none;
  transition: all 250ms;
  border: 0;
  font-size: 16px;
  user-select: none;
  -webkit-user-select: none;
  touch-action: manipulation;
  margin-bottom: 10px;
}

#button:hover {
  box-shadow: rgba(44,187,99,.35) 0 -25px 18px -14px inset,rgba(44,187,99,.25) 0 1px 2px,rgba(44,187,99,.25) 0 2px 4px,rgba(44,187,99,.25) 0 4px 8px,rgba(44,187,99,.25) 0 8px 16px,rgba(44,187,99,.25) 0 16px 32px;
  transform: scale(1.05) rotate(-1deg);
}