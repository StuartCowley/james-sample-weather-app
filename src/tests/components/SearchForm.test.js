import React from "react";
import { render } from "@testing-library/react";
import SearchForm from "../../components/SearchForm";

describe("it renders correctly", () => {
  it("renders a search box and button", () => {
    const { asFragment } = render(<SearchForm />);
    expect(asFragment()).toMatchSnapshot();
  });
});
