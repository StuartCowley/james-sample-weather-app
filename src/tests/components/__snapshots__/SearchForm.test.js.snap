// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly renders a search box and button 1`] = `
<DocumentFragment>
  <div
    class="search-form"
  >
    <input
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Search
    </button>
  </div>
</DocumentFragment>
`;
